joint_state_controller:
  type: joint_state_controller/JointStateController
  # Hz
  publish_rate: 1000
# LEFT ARM
left_SA:
  type: effort_controllers/JointPositionController
  joint: left_SA
  pid: {p: 10000, i: 0, d: 100}
left_SE:
  type: effort_controllers/JointPositionController
  joint: left_SE
  pid: {p: 1000000, i: 0, d: 100}
left_WP:
  type: effort_controllers/JointPositionController
  joint: left_WP
  pid: {p: 10000, i: 0, d: 100}
left_WY:
  type: effort_controllers/JointPositionController
  joint: left_WY
  pid: {p: 3000, i: 0, d: 50}
left_WR:
  type: effort_controllers/JointPositionController
  joint: left_WR
  pid: {p: 100, i: 0, d: 0}
left_linkage_tr:
  type: effort_controllers/JointPositionController
  joint: left_linkage_tr
  pid: {p: 10000, i: 0, d: 100}
left_linkage_tl:
  type: effort_controllers/JointPositionController
  joint: left_linkage_tl
  pid: {p: 1000, i: 0, d: 10}
left_linkage_bl:
  type: effort_controllers/JointPositionController
  joint: left_linkage_bl
  pid: {p: 100, i: 0, d: 1}
# gripper
left_r_outer:
  type: effort_controllers/JointPositionController
  joint: left_r_outer
  pid: {p: 10, i: 0, d: 0}
left_l_outer:
  type: effort_controllers/JointPositionController
  joint: l_outer
  pid: {p: 10, i: 0, d: 0}
left_r_inner:
  type: effort_controllers/JointPositionController
  joint: left_r_inner
  pid: {p: 10, i: 0, d: 0}
left_l_inner:
  type: effort_controllers/JointPositionController
  joint: left_l_inner
  pid: {p: 10, i: 0, d: 0}
left_r_finger:
  type: effort_controllers/JointPositionController
  joint: left_r_finger
  pid: {p: 50, i: 0, d: 0}
left_l_finger:
  type: effort_controllers/JointPositionController
  joint: left_l_finger
  pid: {p: 50, i: 0, d: 0}

# RIGHT ARM
right_SA:
  type: effort_controllers/JointPositionController
  joint: right_SA
  pid: {p: 10000, i: 0, d: 100}
right_SE:
  type: effort_controllers/JointPositionController
  joint: right_SE
  pid: {p: 1000000, i: 0, d: 100}
right_WP:
  type: effort_controllers/JointPositionController
  joint: right_WP
  pid: {p: 10000, i: 0, d: 100}
right_WY:
  type: effort_controllers/JointPositionController
  joint: right_WY
  pid: {p: 3000, i: 0, d: 50}
right_WR:
  type: effort_controllers/JointPositionController
  joint: right_WR
  pid: {p: 100, i: 0, d: 0}
right_linkage_tr:
  type: effort_controllers/JointPositionController
  joint: right_linkage_tr
  pid: {p: 10000, i: 0, d: 100}
right_linkage_tl:
  type: effort_controllers/JointPositionController
  joint: right_linkage_tl
  pid: {p: 1000, i: 0, d: 10}
right_linkage_bl:
  type: effort_controllers/JointPositionController
  joint: right_linkage_bl
  pid: {p: 100, i: 0, d: 1}
# gripper
right_r_outer:
  type: effort_controllers/JointPositionController
  joint: right_r_outer
  pid: {p: 10, i: 0, d: 0}
right_l_outer:
  type: effort_controllers/JointPositionController
  joint: l_outer
  pid: {p: 10, i: 0, d: 0}
right_r_inner:
  type: effort_controllers/JointPositionController
  joint: right_r_inner
  pid: {p: 10, i: 0, d: 0}
right_l_inner:
  type: effort_controllers/JointPositionController
  joint: right_l_inner
  pid: {p: 10, i: 0, d: 0}
right_r_finger:
  type: effort_controllers/JointPositionController
  joint: right_r_finger
  pid: {p: 50, i: 0, d: 0}
right_l_finger:
  type: effort_controllers/JointPositionController
  joint: right_l_finger
  pid: {p: 50, i: 0, d: 0}
